src/snyk/snykOss/services/vulnerabilityCount/npmModuleInfoFetchService.ts

```diff
@@ -0,0 +1,175 @@
+    if (module.string) {
+      npmPackage = this.npmPackageCache.get(module.string) || (await this.fetchNpmRegistryPackage(module));
+      this.npmPackageCache.set(module.string, npmPackage);
+    } else {
+      npmPackage = await this.fetchNpmRegistryPackage(module);
+      this.npmPackageCache.set(module.string, npmPackage);
+    }
+
+    return npmPackage;
+  }
+
+  private async fetchNpmRegistryPackage(module: ImportedModule): Promise<NpmRegistryPackage> {
+    if (module.version && module.name) {
+      return { name: module.name, version: module.version };
+    }
+
+    let dir = this.projectCache.get(module.fileName);
+
+    if (!dir) {
+      const filename = finder(module.fileName).next().filename;
+      if (filename) {
+        dir = dirname(filename);
+        this.projectCache.set(module.fileName, dir);
+      } else {
+        throw new Error('Cannot find "package.json" to determine module key.');
+      }
+    }
+
+    const name = module.name;
+
+    const f = finder(join(dir, 'node_modules', name));
+    const moduleInfo = f.next().value;
+
+    // if the package doesn't start with the package name we were looking for
+    // then it means it's not locally installed, so let's get the version from the
+    // npm registry
+    if (!moduleInfo || !moduleInfo.name || !moduleInfo.name.toLowerCase().startsWith(name)) {
+      try {
+        const { data } = await npmRegistryApi.get<NpmRegistryPackage>(`${name}/latest`);
+
+        return { name, version: data.version };
+      } catch (err) {
+        return {
+          name,
+          version: 'latest',
+        };
+      }
+    }
+
+    return { name: moduleInfo.name, version: moduleInfo.version ?? 'latest' };
+  }
+
+  private lookupVulnerabilities(vulnerabilityKey: string, npmPackageCacheKey: string): Promise<NpmTestApiResult> {
+    const npmPackage = this.npmPackageCache.get(npmPackageCacheKey);
+    if (!npmPackage) {
+      throw new Error('Npm package is not in cache for some reason.');
+    }
+
+    return this.debouncePromise(vulnerabilityKey, this.test.bind(this), npmPackage);
+  }
+
+  private async test(npmPackage: NpmRegistryPackage): Promise<NpmTestApiResult> {
+    const utm = `utm_medium=${this.config.source}&utm_source=${this.config.source}&utm_campaign=${this.config.source}`;
+
+    const result = await npmTestApi.get<NpmTestApiResult>(
+      `/npm/${npmPackage.name}/${npmPackage.version}?${utm}&type=json`,
+    );
+
+    return result.data;
+  }
+
+  private async debouncePromise(
+    name: string,
+    fn: (...args: unknown[]) => Promise<NpmTestApiResult>,
+    ...args: unknown[]
+  ): Promise<NpmTestApiResult> {
+    if (!this.debouncedPromises[name])
+      this.debouncedPromises[name] = _.debounce(
+        async (...args: unknown[]): Promise<NpmTestApiResult> => {
+          try {
+            return await fn(...args);
+          } catch (error) {
+            throw new DebounceError(error);
+          }
+        },
+        2000,
```

Comment:
Why `2000`? Could you make that maybe a constant?

Reply:
Porting Vuln Cost, seems it was working for them. I'll make a constant 👍